NetProbe/Model/NetworkMeasurement.mm

```diff
@@ -53,21 +54,69 @@ @implementation NetworkMeasurement
 
 -(id) init {
     self = [super init];
-    self.logLines = [[NSMutableArray alloc] init];
-    self.finished = FALSE;
+    NSBundle *bundle = [NSBundle mainBundle];
+    geoip_asn = [bundle pathForResource:@"GeoIPASNum" ofType:@"dat"];
+    geoip_country = [bundle pathForResource:@"GeoIP" ofType:@"dat"];
+    ca_cert = [bundle pathForResource:@"cacert" ofType:@"pem"];
+    self.completed = FALSE;
     return self;
 }
 
 -(void) run {
     // Nothing to do here
 }
 
--(NSString*) getDate{
+-(NSString*) getDate {
     NSDateFormatter *dateformatter=[[NSDateFormatter alloc]init];
     [dateformatter setDateFormat:@"dd-MM-yyyy HH:mm:ss"];
     return [dateformatter stringFromDate:[NSDate date]];
 }
 
+-(NSString*) getFileName:(NSString*)ext {
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+    NSString *documentsDirectory = [paths objectAtIndex:0];
+    //double timestamp = mk::time_now();
+    NSString *fileName = [NSString stringWithFormat:@"%@/test-%@.%@", documentsDirectory, self.test_id, ext];
+    //NSLog(@"fileName %@", fileName);
+    return fileName;
+}
+
+-(void)writeOrAppend:(NSString*)string{
+    NSString *fileName = [self getFileName:@"txt"];
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    if(![fileManager fileExistsAtPath:fileName])
+    {
+        [string writeToFile:fileName atomically:NO encoding:NSStringEncodingConversionAllowLossy error:nil];
+    }
+    else
+    {
+        NSFileHandle *myHandle = [NSFileHandle fileHandleForWritingAtPath:fileName];
+        [myHandle seekToEndOfFile];
+        [myHandle writeData:[[NSString stringWithFormat:@"\n%@",string] dataUsingEncoding:NSUTF8StringEncoding]];
+    }
+}
+
+
+- (void)encodeWithCoder:(NSCoder *)coder {
+    [coder encodeObject:self.name forKey:@"Test_name"];
+    [coder encodeObject:self.test_id forKey:@"Test_id"];
+    [coder encodeObject:self.json_file forKey:@"Test_jsonfile"];
+    [coder encodeObject:self.log_file forKey:@"Test_logfile"];
+    [coder encodeObject:[NSNumber numberWithBool:self.completed] forKey:@"test_completed"];
+}
+
+- (id)initWithCoder:(NSCoder *)coder {
+    self = [super init];
+    if (!self) {
+        return nil;
+    }
+    self.name = [coder decodeObjectForKey:@"Test_name"];
+    self.test_id = [coder decodeObjectForKey:@"Test_id"];
+    self.json_file = [coder decodeObjectForKey:@"Test_jsonfile"];
+    self.log_file = [coder decodeObjectForKey:@"Test_logfile"];
+    self.completed = [[coder decodeObjectForKey:@"test_completed"] boolValue];
+    return self;
+}
```

Comment:
Used to save this class on disk (NSUserDefaults)
https://blog.soff.es/archiving-objective-c-objects-with-nscoding